### ISSUE TITLE
Custom slide layout margin isues

### ISSUE THREAD

**Issue by serhii-pokrovskyi** [AppIntro/AppIntro#458]
Hello!
I'm using like this:
activity
```java
 public class FirstRunActivity extends AppIntro2{
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         addSlide(new FirstRunFirstFragment());
 
         setBarColor(Color.parseColor("#3F51B5"));
 
         showSkipButton(true);
         setProgressButtonEnabled(true);
     }
}
```
fragment:

```java
public class FirstRunFirstFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState){
        View view = inflater.inflate(R.layout.fragment_first_run_1, container, false);

        return view;
    }
}
```
fragment_first_run_1 layout:
```xml
<android.support.constraint.ConstraintLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:layout_marginBottom="200dp"
    android:layout_marginTop="200dp">

...............

</android.support.constraint.ConstraintLayout>
```

Expected:
![2017-07-05](https://user-images.githubusercontent.com/10260795/27861719-a57f5f4c-618a-11e7-8f4f-d8ed40128571.png)

Result:
<img src="https://user-images.githubusercontent.com/10260795/27861787-f265c616-618a-11e7-8521-5114b3366ebc.png" width="380">

Question:
How can make fragment layout fill center space only?

**Comment by avluis** [comment]
I see that you are making use of `ConstraintLayout`.
I'm not certain AppIntro has seem much testing with this so it may be an issue to address in the future, but for now, I would recommend taking a look at how the example app implements custom layouts and follow allow.

**Comment by Anutha02** [comment]
You are setting match parent as the layout height of the constraintlayout. If you want the expected result, remove the match_parent attribute and replace it with any value, also try with wrap_content set as the hight.

**Comment by serhii-pokrovskyi** [comment]
> 
> 
> You are setting match parent as the layout height of the constraintlayout. If you want the expected result, remove the match_parent attribute and replace it with any value, also try with wrap_content set as the hight.

Hello, it's been 1.5 after this report =)

Setting wrap_content in my case(1.5 year old case) is not a solution at all.
Main problem is: bottom bar(where exit, next, etc) cover up main content. That is(or was?) a bug, as i see it.

**Comment by Anutha02** [comment]
If that is the case, then it is fixed by #647

**Comment by cortinico** [comment]
This was solved by #647.
I'm closing it
